src/ast.zig

```diff
@@ -1736,3 +1736,17 @@ test smallestEnclosingSubrange {
         children[result4.start .. result4.start + result4.len],
     );
 }
+
+pub fn getReturnTypeNode(tree: Ast, nodes: []const Node.Index) ?Node.Index {
+    var func_buf: [1]Ast.Node.Index = undefined;
+    for (nodes) |node| {
+        const func = tree.fullFnProto(&func_buf, node) orelse continue;
+        return func.ast.return_type.unwrap();
+    }
+    return null;
+}
+
+pub fn getReturnTypeLastToken(tree: Ast, nodes: []const Ast.Node.Index) ?Ast.TokenIndex {
+    const return_type = getReturnTypeNode(tree, nodes) orelse return null;
+    return lastToken(tree, return_type);
+}
```

Comment:
The code in `ast.zig` should be general purpose but these two functions are very specific.

I would suggest to do the following:

- revert changes to analysis.zig
- inline `getReturnTypeNode` into `getReturnTypeLastToken`
- move `getReturnTypeLastToken` to `src/features/completions.zig`